lib/modules/datasource/rubygems/versions-datasource.ts

```diff
@@ -84,72 +144,37 @@ export class VersionsDatasource extends Datasource {
       newLines = (await this.http.get(url, options)).body;
       const durationMs = Math.round(Date.now() - startTime);
       logger.debug(`Rubygems: Fetched rubygems.org versions in ${durationMs}`);
-      regCache.isSupported = true;
     } catch (err) /* istanbul ignore next */ {
       if (err instanceof HttpError && err.response?.statusCode === 404) {
         regCache.isSupported = false;
         return;
       }
-      if (err.statusCode !== 416) {
-        regCache.contentLength = 0;
-        regCache.packageReleases = {};
-        logger.debug({ err }, 'Rubygems fetch error');
-        throw new ExternalHostError(new Error('Rubygems fetch error'));
-      }
-      logger.debug('Rubygems: No update');
-      regCache.lastSync = new Date();
-      return;
-    }
 
-    for (const line of newLines.split(newlineRegex)) {
-      this.processLine(regCache, line);
-    }
-    regCache.lastSync = new Date();
-  }
-
-  private processLine(regCache: RegistryCache, line: string): void {
-    let split: string[] | undefined;
-    let pkg: string | undefined;
-    let versions: string | undefined;
-    try {
-      const l = line.trim();
-      if (!l.length || l.startsWith('created_at:') || l === '---') {
+      if (err.statusCode === 416) {
+        logger.debug('Rubygems: No update');
+        regCache.lastSync = new Date();
         return;
       }
-      split = l.split(' ');
-      [pkg, versions] = split;
-      pkg = VersionsDatasource.copystr(pkg);
-      regCache.packageReleases[pkg] ??= [];
-      const lineVersions = versions.split(',').map((version) => version.trim());
-      for (const lineVersion of lineVersions) {
-        if (lineVersion.startsWith('-')) {
-          const deletedVersion = lineVersion.slice(1);
-          logger.trace({ pkg, deletedVersion }, 'Rubygems: Deleting version');
-          regCache.packageReleases[pkg] = regCache.packageReleases[pkg].filter(
-            (version) => version !== deletedVersion
-          );
-        } else {
-          regCache.packageReleases[pkg].push(
-            VersionsDatasource.copystr(lineVersion)
-          );
-        }
-      }
-    } catch (err) /* istanbul ignore next */ {
-      logger.warn(
-        { err, line, split, pkg, versions },
-        'Rubygems line parsing error'
-      );
+
+      regCache.contentLength = 0;
+      regCache.packageReleases = {};
+
+      logger.debug({ err }, 'Rubygems fetch error');
+      throw new ExternalHostError(err);
```

Comment:
```ts
throw new ExternalHostError(new Error('Rubygems fetch error'));
```

This wasn't too informative, so I provided original `err` value to the `ExternalHostError`